### PR TITLE
Update got and its dependents

### PR DESCRIPTION
**Pull request by shinnn**
## Purpose of this pull request? 

- [ ] Documentation update
- [ ] Bug fix 
- [x] Enhancement
- [ ] Other, please explain:

### Problem

`yo` has [got](https://github.com/sindresorhus/got) v5 in its deep dependencies:

```
$ npm ls got
yo@2.0.0 /Users/shinnn/github/yo
├── got@6.7.1 
├─┬ npm-keyword@4.2.0
│ └── got@5.7.1 
├─┬ package-json@2.4.0
│ └── got@5.7.1 
└─┬ update-notifier@2.2.0
  └─┬ latest-version@3.1.0
    └─┬ package-json@4.0.1
      └── got@6.7.1  deduped
```

The problem is that got v5 has an [`engine` restriction `"node": ">=0.10.0 <7"`](https://github.com/sindresorhus/got/blob/856b4caf16b02ce28ef0d92e83cf434a50b71e84/package.json#L20) that prevents the latest Node.js from installing it in [`engine-strict` mode](https://docs.npmjs.com/misc/config#engine-strict). 

```console
$ cat .npmrc
engine-strict=true

$ npm i yo
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for got@5.7.1: wanted: {"node":">=0.10.0 <7"} (current: {"node":"9.4.0","npm":"5.6.0"})
npm ERR! notsup Not compatible with your version of node/npm: got@5.7.1
npm ERR! notsup Not compatible with your version of node/npm: got@5.7.1
npm ERR! notsup Required: {"node":">=0.10.0 <7"}
npm ERR! notsup Actual:   {"npm":"5.6.0","node":"9.4.0"}
```

### Solution

* I updated [`npm-keyword`](https://github.com/sindresorhus/npm-keyword) and [`package-json`](https://github.com/sindresorhus/package-json) that depend on older `got`.
* I also updated `got` to the latest version.

## Is there anything you'd like reviewers to focus on?

I'll leave comments to notable changes.

### CODE REVIEW

File: lib/routes/install.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ function searchMatchingGenerators(app, term, cb) {
       }));
     }, cb);
   }
-  got('yeoman.io/blacklist.json', {json: true})
+  got('http://yeoman.io/blacklist.json', {json: true})
```
Comment (shinnn) [inline lib/routes/install.js]: https://github.com/sindresorhus/got/releases/tag/v8.0.0

> [Breaking change] Uses HTTPS for protocol-less URLs (Previously it defaulted to http)

Currently `yeoman.io` doesn't seem to support HTTPS. https://yeoman.io

**Comment by shinnn** [comment]
Sorry, I found almost the same PR https://github.com/yeoman/yo/pull/549 after submitting this. This PR passes all the tests that #549 fails, though.

File: test/route-install.js [hunk 2]
```diff
@@ -46,15 +67,20 @@ describe('install route', () => {
       ];
 
       this.pkgData = {
-        author: {
-          name: 'Simon'
+        'dist-tags': {
```
Comment (shinnn) [inline test/route-install.js]: The latest `package-json` expects these properties: https://github.com/sindresorhus/package-json/blob/2312d0437c16d7117d354ae76f640c5ac25fe0e0/index.js#L39-L56

**Comment by SBoudrias** [comment]
Thanks a lot for the PR!

I'll try to cut a release tonight or tomorrow.

**Comment by shinnn** [comment]
@SBoudrias No problem. Thanks for [the new release](https://github.com/yeoman/yo/commit/6153a0d400af17744ab419fbe98118b0086c0c8b).

**Comment by mischah** [comment]
@shinnn 

> @mischah, sorry, I found almost the same PR #549 after submitting this. Mine passes all the tests that #549 fails, though.

Thank you so much :octocat:

**Pull request merged**